### ISSUE TITLE
Fix broken links in the docs

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#2380]
Thanks to @astrofrog on twitter I learned that if you run:

```
$ make linkcheck
```

from inside the `docs/` directory you get a report of the status of all the URLs in your docs. When I ran this I got the following broken links:

```
(line   18) broken    http://hesperia.gsfc.nasa.gov/ssw/hessi/doc/guides/hessi_data_access.htm#Observing - Anchor 'Observing' not found
(line   18) broken    http://hesperia.gsfc.nasa.gov/ssw/hessi/doc/guides/hessi_data_access.htm#Observing - Anchor 'Observing' not found
(line   32) broken    http://gammaray.nsstc.nasa.gov/gbm/ - HTTPSConnectionPool(host='gammaray.nsstc.nasa.gov', port=443): Max retries exceeded with url: /gbm/ (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   35) broken    http://gammaray.msfc.nasa.gov/gbm/publications/ - HTTPSConnectionPool(host='gammaray.nsstc.nasa.gov', port=443): Max retries exceeded with url: /gbm/publications/ (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   25) broken    http://legacy-www.swpc.noaa.gov/Data/index.html#indices - Anchor 'indices' not found
(line   20) broken    http://legacy-www.swpc.noaa.gov/Data/index.html#indices - Anchor 'indices' not found
(line   13) broken    http://secchi.nrl.navy.mil - HTTPSConnectionPool(host='secchi.nrl.navy.mil', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   15) broken    http://secchi.nrl.navy.mil/index.php?p=cor2 - HTTPSConnectionPool(host='secchi.nrl.navy.mil', port=443): Max retries exceeded with url: /index.php?p=cor2 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   16) broken    http://cor1.gsfc.nasa.gov/guide/ - 404 Client Error: Not Found for url: https://stereo.gsfc.nasa.gov/guide/
(line   11) broken    http://secchi.nrl.navy.mil - HTTPSConnectionPool(host='secchi.nrl.navy.mil', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   15) broken    http://secchi.nrl.navy.mil - HTTPSConnectionPool(host='secchi.nrl.navy.mil', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   17) broken    http://lasco-www.nrl.navy.mil/index.php?p=content/keywords - HTTPSConnectionPool(host='lasco-www.nrl.navy.mil', port=443): Max retries exceeded with url: /index.php?p=content/keywords (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   16) broken    http://lasco-www.nrl.navy.mil - HTTPSConnectionPool(host='lasco-www.nrl.navy.mil', port=443): Max retries exceeded with url: / (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   18) broken    http://lasco-www.nrl.navy.mil/index.php?p=content/handbook/hndbk - HTTPSConnectionPool(host='lasco-www.nrl.navy.mil', port=443): Max retries exceeded with url: /index.php?p=content/handbook/hndbk (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   13) broken    http://solarb.msfc.nasa.gov/index.html - HTTPSConnectionPool(host='hinode.msfc.nasa.gov', port=443): Max retries exceeded with url: /index.html (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   14) broken    http://proba2.oma.be/index.html/swap/swap-analysis-manual/article/data-products?menu=23 - 404 Client Error: Not Found for url: http://proba2.oma.be/index.html/swap/swap-analysis-manual/article/data-products?menu=23
(line   20) broken    http://proba2.oma.be/index.html/swap/swap-analysis-manual/article/data-products?menu=23 - 404 Client Error: Not Found for url: http://proba2.oma.be/index.html/swap/swap-analysis-manual/article/data-products?menu=23
(line   12) broken    http://solarb.msfc.nasa.gov/index.html - HTTPSConnectionPool(host='hinode.msfc.nasa.gov', port=443): Max retries exceeded with url: /index.html (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   15) broken    http://hinode.nao.ac.jp/hsc_e/instruction_e.shtml - 404 Client Error: Not Found for url: http://hinode.nao.ac.jp/hsc_e/instruction_e.shtml
(line   14) broken    http://hinode.nao.ac.jp/sbsc/SB_MW_Key13.pdf - 404 Client Error: Not Found for url: http://hinode.nao.ac.jp/sbsc/SB_MW_Key13.pdf
(line   33) broken    http://gammaray.nsstc.nasa.gov/gbm/ - HTTPSConnectionPool(host='gammaray.nsstc.nasa.gov', port=443): Max retries exceeded with url: /gbm/ (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   36) broken    http://gammaray.msfc.nasa.gov/gbm/publications/ - HTTPSConnectionPool(host='gammaray.nsstc.nasa.gov', port=443): Max retries exceeded with url: /gbm/publications/ (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line   27) broken    http://legacy-www.swpc.noaa.gov/Data/index.html#indices - Anchor 'indices' not found
(line   23) broken    http://legacy-www.swpc.noaa.gov/Data/index.html#indices - Anchor 'indices' not found
(line  414) broken    https://raw.githubusercontent.com/astropy/astropy/master/licenses/LICENSE.rst - 404 Client Error: Not Found for url: https://raw.githubusercontent.com/astropy/astropy/master/licenses/LICENSE.rst
(line    3) broken    http://hesperia.gsfc.nasa.gov/ssw/trace/idl/util/routines/tr_get_disp.pro - 404 Client Error: Not Found for url: https://hesperia.gsfc.nasa.gov/ssw/trace/idl/util/routines/tr_get_disp.pro
(line    9) broken    http://books.google.com/books?id=4SWENr1tIJ0C&printsec=frontcover&source=gbs_ge_summary_r&cad=0#v=onepage&q=sfu&f=false - Anchor 'v=onepage&q=sfu&f=false' not found
(line   12) broken    https://riot.im/app/#/room/#sunpy-general:matrix.org - Anchor '/room/#sunpy-general:matrix.org' not found
(line   27) broken    http://docs.sunpy.org/en/latest/dev.html - 404 Client Error: Not Found for url: http://docs.sunpy.org/en/latest/dev.html
(line   53) broken    https://conda.io/docs/install/quick.html#quick-install - Anchor 'quick-install' not found
(line   88) broken    https://riot.im/app/#/room/#sunpy-general:matrix.org - Anchor '/room/#sunpy-general:matrix.org' not found
(line   30) broken    http://blogs.discovermagazine.com/badastronomy/2011/06/07/the-sun-lets-loose-a-huge-explosion/#.WVvrBsaZORs - Anchor '.WVvrBsaZORs' not found
(line    8) broken    http://www.google-melange.com/gsoc/homepage/google/gsoc2013 - HTTPSConnectionPool(host='www.google-melange.com', port=443): Max retries exceeded with url: /gsoc/homepage/google/gsoc2013 (Caused by SSLError(SSLError("bad handshake: Error([('SSL routines', 'ssl3_get_server_certificate', 'certificate verify failed')],)",),))
(line  123) broken    http://www.scipy.org/Tentative_NumPy_Tutorial#head-864862d3f2bb4c32f04260fac61eb4ef34788c4c - Anchor 'head-864862d3f2bb4c32f04260fac61eb4ef34788c4c' not found
(line  151) broken    http://www.pip-installer.org/en/1.2.1/other-tools.html#using-pip-with-the-user-scheme - Anchor 'using-pip-with-the-user-scheme' not found
```

**Comment by Cadair** [comment]
It looks like the Riot ones there will always fail. I don't think it will be possible to get that list to zero. Still it would be worth updating them where possible.

**Comment by wafels** [comment]
Must be pretty strict.  Some of links when clicked on automatically
re-direct to their https versions.

**Comment by nabobalis** [comment]
Well redirects can be dangerous.

We could fix most of those. Something for someone to spend some time in.

**Comment by CaptainDaVinci** [comment]
Redirected links should also be fixed or just the broken links?

**Comment by nabobalis** [comment]
Go with both.

**Comment by cmolinaord** [comment]
Hi. I'm working on this issue. I've fixed the redirected links, and I may spend a bit more time with the broken ones, because sometimes I have to google to find the page that is intended to be linked.
You can see my progress (only the redirects) in my fork. And when I'll finish, I'll do the PR.

**Comment by cmolinaord** [comment]
Hi! I'm having a problem when trying to fixing this . To see what I've already changed I run make linkcheck in docs/ but as a result I obtain redirects and broken links that I've already fixed (and they don't appear in the project). Do you know what it's happening? Do I need to re-compile something?

**Comment by dpshelio** [comment]
@cmolinaord feel free to create a PR as soon as you start working on this, so we can help you straight away.

**Comment by nabobalis** [comment]
Fixed mostly by #2403. 

Having had to work with this, I think it's a losing battle to add this to the CI. It might be something we want to check before each major release tho.
